src/request.rs

```diff
@@ -306,6 +311,57 @@ impl SecretType for SecretConnection {
     }
 }
 
+struct SecretResource {
+    request: RequestDetails,
+}
+
+#[async_trait]
+impl SecretType for SecretResource {
+    async fn payload(&self, db: &KbsDb, _connection: Connection) -> Result<Vec<u8>> {
+        // For now all resources must be in the top level of the
+        // resources directory.
+        let resource_path = db.get_resource_path(&self.request.id).await?;
+
+        let mut path = PathBuf::from(RESOURCE_PATH_BASE);
+        path.push(resource_path);
+
+        let resource_binary = std::fs::read(path.as_path())?;
+
+        match &self.request.format[..] {
+            "binary" => Ok(resource_binary),
+            "json" => {
+                let json_payload =
+                    json!({ self.request.id.clone(): base64::encode(resource_binary) });
+                Ok(bincode::serialize(&json_payload)?)
+            }
+            _ => Err(anyhow!("Unknown Secret Format")),
+        }
+    }
+
+    async fn policies(&self, db: &KbsDb) -> Vec<policy::Policy> {
+        match db.get_signing_keys_policy(&self.request.id).await {
+            Ok(policy) => match policy {
+                Some(p) => vec![p],
+                None => {
+                    info!("No policy for resource: {}", &self.request.id);
+                    vec![]
+                }
+            },
```

Comment:
(future) This appears in several places in our code, maybe extract a helper function.